builder/evaluator.go

```diff
@@ -169,12 +169,19 @@ func (b *Builder) Run(context io.Reader) (string, error) {
 
 // Reads a Dockerfile from the current context. It assumes that the
 // 'filename' is a relative path from the root of the context
-func (b *Builder) readDockerfile(filename string) error {
-	filename = path.Join(b.contextPath, filename)
+func (b *Builder) readDockerfile(origFile string) error {
+	filename := filepath.Join(b.contextPath, origFile)
+
+	tmpDockerPath := filepath.Dir(filename) + string(os.PathSeparator)
+	tmpContextPath := filepath.Clean(b.contextPath) + string(os.PathSeparator)
```

Comment:
I'm not joining a path and a name - I'm just making sure that these two paths end with /'s